taqueria-vscode-extension/src/lib/helpers.ts

```diff
@@ -328,6 +330,46 @@ export const inject = (deps: InjectedDependencies) => {
 			);
 		};
 
+	const updateCommandStates = async (
+		context: api.ExtensionContext,
+		output: api.OutputChannel,
+		i18n: i18n,
+		projectDir: Util.PathToDir,
+	) => {
+		try {
+			const config = await Util.TaqifiedDir.create(projectDir, i18n);
+			vscode.commands.executeCommand('setContext', '@taqueria-state/is-taqified', !!config.config);
+			const plugins = getWellKnownPlugins();
+			for (const plugin of plugins) {
+				const found = config.config.plugins?.find(item => item.name === plugin) !== undefined;
+				vscode.commands.executeCommand('setContext', plugin, found);
+			}
+		} catch (e: any) {
+			// After logging PR is merged, log this error
+		}
+	};
+
+	const createWatcherIfNotExists = (
+		context: api.ExtensionContext,
+		output: api.OutputChannel,
+		i18n: i18n,
+		projectDir: Util.PathToDir,
+		createConfigWatcherIfNotExists: (folder: string, factory: () => api.FileSystemWatcher) => void,
+	) => {
+		createConfigWatcherIfNotExists(projectDir, () => {
```

Comment:
This is a function passed as an argument to `createWatcherIfNotExists`

1. Why is `createConfigWatcherIfNotExists` an argument? Can't it be a normal function?

Reply:
Because that functionality is implemented in `extension.ts`. That file depends on `helpers.ts`. If I call that method directly, I will also depend `helpers.ts` on `extension.ts`. Regardless of whether or not this is supported by our project system, this is a cyclic dependency that is better be avoided.

Reply:
In general passing functions as arguments is not a bad thing. Lots of functionality like `map` and `filter` take functions as arguments. But whether or not this specific implementation could be done better can be a good water-cooler conversation for knowledge sharing.

Reply:
Yes but this argument seems very specific, just by looking at the name: `createConfigWatcherIfNotExists`. I am under the impression that only one function value will ever be passed to `createWatcherIfNotExists`. If that's the case, it sounds like a code smell.